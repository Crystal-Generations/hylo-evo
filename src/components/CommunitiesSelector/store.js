import { createSelector } from 'reselect'
import sampleCommunities from './sampleCommunities'

const FIND_COMMUNITIES = 'hyloEditor/FIND_COMMUNITIES'
const CLEAR_COMMUNITIES = 'hyloEditor/CLEAR_COMMUNITIES'

// Action Creators

export function findCommunities (searchText) {
  return {
    type: FIND_COMMUNITIES,
    payload: { searchText }
  }
}

export function clearCommunities (searchText) {
  return { type: CLEAR_COMMUNITIES }
}

// Reducer

const defaultState = {
  communitiesResults: sampleCommunities
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_COMMUNITIES:
      // filter(i =>
      //   String(i
      //     .get('name')
      //     .replace(/\s+/g, '')
      //     .toLowerCase()
      //     .indexOf(lowerSearch) !== -1
      //   )
      // )
      // .take(5)
      return {...state, communitiesResults: filter(payload.searchText)(sampleCommunities)}
    case CLEAR_COMMUNITIES:
      return {...state, communitiesResults: []}
    default:
      return state
  }
}

// Selectors

export const getCommunitiesResults = createSelector(
  (state) => state.CommunitiesSelector,
  (state, props) => state.communitiesResults
)
