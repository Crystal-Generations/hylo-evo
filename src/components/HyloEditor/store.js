import { createSelector } from 'reselect'
import { fromJS } from 'immutable'
import sampleMentions from './sampleMentions'
import mentionPlugin from 'draft-js-mention-plugin'
import hashtagPlugin from './hashtagPlugin'

const defaultMentionsSuggestionFilter = mentionPlugin.defaultSuggestionsFilter
const defaultHashtagSuggestionFilter = hashtagPlugin.defaultSuggestionsFilter

const FIND_MENTIONS = 'hyloEditor/FIND_MENTIONS'
const CLEAR_MENTIONS = 'hyloEditor/CLEAR_MENTIONS'
const FIND_HASHTAGS = 'hyloEditor/FIND_HASHTAGS'
const CLEAR_HASHTAGS = 'hyloEditor/CLEAR_HASHTAGS'

// Action Creators

export function findMentions (searchText) {
  return {
    type: FIND_MENTIONS,
    payload: { searchText }
  }
}

export function clearMentions (searchText) {
  return { type: CLEAR_MENTIONS }
}

export function findHashtags (searchText) {
  return {
    type: FIND_HASHTAGS,
    payload: { searchText }
  }
}

export function clearHashtags (searchText) {
  return { type: CLEAR_HASHTAGS }
}

// Reducer

export default function reducer (state = {mentionResults: fromJS([])}, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_MENTIONS:
      return {...state, mentionResults: defaultMentionsSuggestionFilter(payload.searchText, fromJS(sampleMentions))}
    case CLEAR_MENTIONS:
      return {...state, mentionResults: fromJS([])}
    case FIND_HASHTAGS:
      return {...state, mentionResults: defaultHashtagSuggestionFilter(payload.searchText, fromJS(['test', 'test2']))}
    case CLEAR_HASHTAGS:
      return {...state, mentionResults: fromJS([])}
    default:
      return state
  }
}

// Selectors

export const getMentionResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.mentionResults
)

export const getHashtagResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.hasTagResults
)
