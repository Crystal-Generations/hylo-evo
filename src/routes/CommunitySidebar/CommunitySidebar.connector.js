import { connect } from 'react-redux'
import getParam from 'store/selectors/getParam'
import getMe from 'store/selectors/getMe'
import getCommunityForCurrentRoute from 'store/selectors/getCommunityForCurrentRoute'
import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'
import { find, get } from 'lodash/fp'

const getCanModerate = ormCreateSelector(
  orm,
  state => state.orm,
  (state, props) => props.community,
  ({ Membership }, community) => {
    const memberships = Membership.all().toRefArray()
    const membership = find(m =>
      m.community === get('id', community), memberships)
    return get('hasModeratorRole', membership)
  }
)

export function mapStateToProps (state, props) {
  const community = getCommunityForCurrentRoute(state, props)
  const members = community ? community.members.toModelArray().slice(0, 8) : []
  const leaders = community ? community.moderators.toModelArray() : []
  const currentUser = getMe(state, props)

  const canModerate = getCanModerate(state, {community})

  return {
    community: community ? community.ref : null,
    members,
    leaders,
    slug: getParam('slug', state, props),
    currentUser,
    canModerate
  }
}

export default connect(mapStateToProps)
