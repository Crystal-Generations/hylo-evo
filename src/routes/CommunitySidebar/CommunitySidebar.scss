$about-height: 116px;

// CommunitySidebar

.community-sidebar {
  padding-top: $space-6x;
}

// AboutSection

.about-section {
  margin-top: $space-3x;
}

.about-header {
  composes: bdy-drk-lg from 'css/typography.scss';
  margin-bottom: $space-4x;
}

.about {
  composes: caption-lt-lg from 'css/typography.scss';
  max-height: $about-height;
  overflow: hidden;
  margin-bottom: 5px;
}

.gradient {
  position: absolute;
  background-image: linear-gradient(0, white, transparent 50%, transparent);
  height: $about-height;
  width: 100%;
}

.expanded {
  max-height: none;
}

.expand-button {
  composes: text-button from 'css/typography.scss';
}
