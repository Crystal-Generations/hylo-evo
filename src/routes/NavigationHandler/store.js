const NAVIGATE = 'NAVIGATE'
const RESET_NAVIGATION = 'RESET_NAVIGATION'

export function navigate (to) {
  return {
    type: NAVIGATE,
    payload: {to}
  }
}

export function resetNavigation (to) {
  return {type: RESET_NAVIGATION}
}

const reducer = (state = null, { type, error, payload, meta }) => {
  switch (type) {
    case NAVIGATE:
      return payload.to
    case RESET_NAVIGATION:
      return null
  }
  return state
}

export default reducer
