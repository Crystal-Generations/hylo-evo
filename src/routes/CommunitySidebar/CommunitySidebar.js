import React, { PropTypes, Component } from 'react'
import Loading from 'components/Loading'
import './CommunitySidebar.scss'
const { object, shape } = PropTypes

export default class CommunitySidebar extends Component {
  static propTypes = {
    commmunity: object
  }

  render () {
    const { commmunity } = this.props
    if (!commmunity) return <Loading />
    const { name } = commmunity

    return <div>
      <div styleName='about-header'>
        About {name}
      </div>
    </div>
  }
}
