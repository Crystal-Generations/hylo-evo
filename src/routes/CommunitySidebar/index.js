import CommunitySidebar from './CommunitySidebar'
import connector from './CommunitySidebar.connector'
import { HOLOCHAIN_ACTIVE } from 'util/holochain'

export default HOLOCHAIN_ACTIVE
  ? null
  : connector(CommunitySidebar)
