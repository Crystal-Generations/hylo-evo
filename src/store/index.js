import React from 'react'
import { createStore, applyMiddleware, combineReducers, compose } from 'redux'
import { Provider } from 'react-redux'
import { ConnectedRouter, routerReducer, routerMiddleware } from 'react-router-redux'
import createHistory from 'history/createBrowserHistory'
import { compact, flatten } from 'lodash'
import reducers from '../reducers'
import routes from '../routes'

const history = createHistory()

// const middleware = compact(flatten([
//   routerMiddleware(history)
// ]))

const middleware = routerMiddleware(history)

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(
  combineReducers({
    // ...reducers,
    router: routerReducer
  }),
  composeEnhancers(
    applyMiddleware(middleware)
  )
)

const app = <Provider store={store}>
  <ConnectedRouter history={history}>
    {routes}
  </ConnectedRouter>
</Provider>

export default app
