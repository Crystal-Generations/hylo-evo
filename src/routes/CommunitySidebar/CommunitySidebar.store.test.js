import orm from 'store/models' // this initializes redux-orm
import { getCanModerate } from './CommunitySidebar.store'

describe('getCanModerate', () => {
  const session = orm.session(orm.getEmptyState())
  beforeEach(() => {
    session.Me.create({
      id: '1'
    })
  })
  it('returns expected values', () => {
    const state = { orm: session.state }

    const props = {
      community: {
        id: 1,
        hasModeratorRole: true
      }
    }
    expect(getCanModerate(state, props)).toMatchSnapshot()
  })
})
