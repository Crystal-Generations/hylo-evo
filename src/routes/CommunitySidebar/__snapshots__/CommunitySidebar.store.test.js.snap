// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCanModerate returns expected values 1`] = `undefined`;
